packages/fhir-location-management/src/components/LocationForm/utils.ts

```diff
@@ -125,6 +125,8 @@ export const generateLocationUnit = (
 
   if (id) {
     payload.id = id;
+  } else {
+    payload.id = v4();
   }
```

Comment:
```suggestion
payload.id = id;
  if (!id) {
     payload.id = v4();
  }
```

Reply:
Nitpick

Reply:
I wont be editing this, given its a nitpick and the logic is still sound and I think its easier to follow